Automattic/sensei
# Add Course List block to course archive on page creation

Implements https://github.com/Automattic/sensei/issues/6345

### Changes proposed in this Pull Request

* Added Course List block in Course Archive page

### Testing instructions

- Make sure you have an existing installation of Sensei having this branch checked out; keep it as it is
- Create a fully new local WP site containing this branch
- Install the Sensei Plugin in the new installation and complete the onboarding
- Create multiple Courses and publish them
- Go to the Courses page from frontend
- Make sure you see courses rendered by the Course List block
- Go to wp-admin->Pages->Courses, change something in the course List block, and save.
- Make sure the change takes effect in frontend
- Now delete the Course List block from the page using the editor
- Check the Courses page from frontend again, and make sure the old archive page is being rendered

### New/Updated Hooks

`sensei_course_archive_page_template` - Filters contents for the archive page being created.

### Screenshot / Video

![Screen Shot 2023-01-07 at 4 33 05 AM](https://user-images.githubusercontent.com/6820724/211113317-5982dd0b-b762-4f41-84fc-03319430e859.png)
